src/prepared_statement.cpp

```diff
@@ -98,7 +98,7 @@ namespace sqlpp
       param.buffer = const_cast<uint64_t*>(value);
       param.buffer_length = sizeof(*value);
       param.length = &param.buffer_length;
-      param.is_null = &_handle->stmt_param_is_null[index];
+      param.is_null = (my_bool*)is_null;
```

Comment:
Why did you change this? c-style casting a bool to a pointer looks rather dangerous to me (even more so, considering that `is_null` is a temporary).

Reply:
I had changed this because the compiler was erroring out on that line, saying that 
> &_handle->stmt_param_is_null[index]
was temporary. I don't agree with that statement but this was just a temp change while I tested the rest of the comp process.

Reply:
this is the error it throws:

>sqlpp11-connector-mysql/src/prepared_statement.cpp:69:57: error: cannot convert ‘std::vector<bool>::reference* {aka std::_Bit_reference*}’ to ‘bool*’ in assignment
sqlpp11-connector-mysql/src/prepared_statement.cpp: In member function ‘void sqlpp::mysql::prepared_statement_t::_bind_integral_parameter(size_t, const int64_t*, bool)’:
sqlpp11-connector-mysql/src/prepared_statement.cpp:85:57: error: taking address of temporary [-fpermissive]
       param.is_null = &_handle->stmt_param_is_null[index];

Reply:
Alright just found out that the reason why is because std::vector can not do what we are doing with booleans. that call instead of returning the address of the bool, it returns the address of a proxy object that IS temporary.

https://en.cppreference.com/w/cpp/container/vector_bool

Reply:
Aaargh, yes, it worked before because of `my_bool != bool`.

I am confused by all the `reinterpret_cast`? I think we should just put something different into the vector, like a struct with just a my_bool/bool member.